### PR TITLE
Check if alpha is not None when restricting it to be at most 1

### PR DESCRIPTION
**Pull request by panpan2**
## Types of changes
scipy.optimize.linesearch.scalar_search_armijo can return alpha=None ([function definition](https://github.com/scipy/scipy/blob/master/scipy/optimize/linesearch.py#L689)). In optim.py:line_search_armijo as part of a previous issue #184, we restrict the value of alpha to be at most 1 with min(1, alpha) but this throws an Exception when alpha is None. In this PR, we just check the value of `alpha` is not `None` when making the comparison with `1`.

- [ ] Docs change / refactoring / dependency upgrade
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Motivation and context / Related issue

Issue: [https://github.com/PythonOT/POT/issues/198](https://github.com/PythonOT/POT/issues/198)

## Checklist

- [x] The documentation is up-to-date with the changes I made.
- [x] I have read the [**CONTRIBUTING**](CONTRIBUTING.md) document.
- [x] All tests passed, and additional code has been covered with new tests.

### CODE REVIEW

File: ot/optim.py [hunk 1]
```diff
@@ -69,7 +69,7 @@ def phi(alpha1):
     alpha, phi1 = scalar_search_armijo(
         phi, phi0, derphi0, c1=c1, alpha0=alpha0)
 
-    return min(1, alpha), fc[0], phi1
+    return min(1, alpha) if alpha is not None else 1, fc[0], phi1
```
Comment (rflamary) [inline ot/optim.py:72]: could you put the test in another line with a small comment about what is necessary?

It is a bit hard to follow in the return

Comment (rflamary) [reply ot/optim.py:72]: Also I guess you have an example where the original function failed. Could you do a simple no regression test it test_optim.py to check that this problem never re-appears again? It feels as if this function always needs tuning ;)

Comment (rflamary) [reply ot/optim.py:72]: Hello @panpan2 could you please take into account my comment so that we can merge your PR?

Comment (panpan2) [reply ot/optim.py]: Updated! :)

Comment (panpan2) [reply ot/optim.py]: @rflamary any other changes needed to merge this?
